src/plugins/discover/public/application/components/context_app/use_context_app_fetch.test.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { act, renderHook } from '@testing-library/react-hooks';
+import { setServices, getServices } from '../../../kibana_services';
+import { SortDirection } from '../../../../../data/public';
+import { createFilterManagerMock } from '../../../../../data/public/query/filter_manager/filter_manager.mock';
+import { CONTEXT_TIE_BREAKER_FIELDS_SETTING } from '../../../../common';
+import { DiscoverServices } from '../../../build_services';
+import { indexPatternMock } from '../../../__mocks__/index_pattern';
+import { indexPatternsMock } from '../../../__mocks__/index_patterns';
+import { FailureReason, LoadingStatus } from '../../angular/context_query_state';
+import { ContextAppFetchProps, useContextAppFetch } from './use_context_app_fetch';
+import {
+  mockAnchorHit,
+  mockPredecessorHits,
+  mockSuccessorHits,
+} from './__mocks__/use_context_app_fetch';
+
+const mockFilterManager = createFilterManagerMock();
+
+jest.mock('../../angular/context/api/context', () => ({
+  fetchContextProvider: () => ({
+    fetchSurroundingDocs: (type: string, indexPatternId: string) => {
+      if (!indexPatternId) {
+        throw new Error();
+      }
+      return type === 'predecessors' ? mockPredecessorHits : mockSuccessorHits;
+    },
+  }),
+}));
+
+jest.mock('../../angular/context/api/anchor', () => ({
+  fetchAnchorProvider: () => (indexPatternId: string) => {
+    if (!indexPatternId) {
+      throw new Error();
+    }
+    return mockAnchorHit;
+  },
+}));
+
+const initDefaults = (tieBreakerFields: string[], indexPatternId = 'the-index-pattern-id') => {
+  const dangerNotification = jest.fn();
+
+  setServices(({
+    data: {
+      search: {
+        searchSource: {
+          createEmpty: jest.fn(),
+        },
+      },
+    },
+    indexPatterns: indexPatternsMock,
+    toastNotifications: { addDanger: dangerNotification },
+    core: { notifications: { toasts: [] } },
+    history: () => {},
+    filterManager: mockFilterManager,
+    uiSettings: {
+      get: (key: string) => {
+        if (key === CONTEXT_TIE_BREAKER_FIELDS_SETTING) {
+          return tieBreakerFields;
+        }
+      },
+    },
+  } as unknown) as DiscoverServices);
+
+  return {
+    dangerNotification,
+    props: ({
+      anchorId: 'mock_anchor_id',
+      indexPatternId,
+      indexPattern: indexPatternMock,
+      appState: {
+        sort: [['order_date', SortDirection.desc]],
+        predecessorCount: 2,
+        successorCount: 2,
+      },
+      useNewFieldsApi: false,
```

Comment:
can this be set to true?

Reply:
It can be set to true but it not really affect the resault of the tests since the entire `'../../angular/context/api/context'` and `'../../angular/context/api/anchor'` modules were mocked in the top of the file and will return hits not repending on the `useNewFieldsApi`